From 367d3289ee3e0a51cc7a7a7422d4b497bce47985 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 28 Jun 2022 18:58:11 +0200
Subject: [PATCH 099/954] fix(st-clock): force SRAM1 and SRAM2 clock enable

SRAM1 and SRAM2 is used in BL2, they should marked as CRITICAL.

SRAM1 is already enable in ROM code but not SRAM2, used in TF-A BL2
to load DDR FIP in case of serial boot.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 48fe88151000d4a0537f823a015f1ba9d0fe5160 fix(st-clock): force SRAM1 and SRAM2 clock enable]
Change-Id: Icf9a59d98101f5ee755dc36135ab0da8e95e664e
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/258223
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/clk/stm32mp2_clk.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/st/clk/stm32mp2_clk.c b/drivers/st/clk/stm32mp2_clk.c
index 66c414aaf6..8857d7d1b5 100644
--- a/drivers/st/clk/stm32mp2_clk.c
+++ b/drivers/st/clk/stm32mp2_clk.c
@@ -997,8 +997,8 @@ static const struct clk_stm32 stm32mp25_clk[CK_LAST] = {
 	/* KERNEL CLOCK */
 	GATE(_CK_SYSRAM, CK_BUS_SYSRAM, _CK_ICN_HS_MCU, 0, RCC_SYSRAMCFGR, 1),
 	GATE(_CK_RETRAM, CK_BUS_RETRAM, _CK_ICN_HS_MCU, 0, RCC_RETRAMCFGR, 1),
-	GATE(_CK_SRAM1, CK_BUS_SRAM1, _CK_ICN_HS_MCU, 0, RCC_SRAM1CFGR, 1),
-	GATE(_CK_SRAM2, CK_BUS_SRAM2, _CK_ICN_HS_MCU, 0, RCC_SRAM2CFGR, 1),
+	GATE(_CK_SRAM1, CK_BUS_SRAM1, _CK_ICN_HS_MCU, CLK_IS_CRITICAL, RCC_SRAM1CFGR, 1),
+	GATE(_CK_SRAM2, CK_BUS_SRAM2, _CK_ICN_HS_MCU, CLK_IS_CRITICAL, RCC_SRAM2CFGR, 1),
 
 	GATE(_CK_DDRPHYC, CK_BUS_DDRPHYC, _CK_ICN_LS_MCU, 0, RCC_DDRPHYCAPBCFGR, 1),
 	GATE(_CK_SYSCPU1, CK_BUS_SYSCPU1, _CK_ICN_LS_MCU, 0, RCC_SYSCPU1CFGR, 1),
-- 
2.53.0

